src/parser/transform/tableref/transform_table_function.cpp

```diff
@@ -28,6 +28,20 @@ unique_ptr<TableRef> Transformer::TransformRangeFunction(duckdb_libpgquery::PGRa
 	case duckdb_libpgquery::T_PGFuncCall: {
 		auto func_call = PGPointerCast<duckdb_libpgquery::PGFuncCall>(call_tree.get());
 		result->function = TransformFuncCall(*func_call);
+		std::string function_name =
+		    PGPointerCast<duckdb_libpgquery::PGValue>(func_call->funcname->head->data.ptr_value)->val.str;
+		// temporary, ugly placeholder
+		std::vector<std::string> allowed_functions {"unnest", "repeat", "repeat_row",    "generate_series",
```

Comment:
I think we want to add a flag to the TableFunction to communicate whether it supports ordinality instead, but that might be your plan already? Since this is marked as a placeholder

Reply:
Adding this Flag to the **TableFunction** with the _default = false_ , and then setting the Flag in the constructor of the specific functions would probably be the best option. 
However, at this point during execution I only have access to a **TableFunctionRef**, which contains a pointer to a **FunctionExpression.**
The Problem with this **FunctionExpression** is that it only contains the name of the function as string.

One option would be to include the flag in the **TableFunctionRef**, which is also the object where I included a bool for propagating whether or not _WITH ORDINALITY_ has been requested in the parser. The  problem here is that I struggle to find the place where I could set this flag depending on the concrete Function being called, and would probably need a hint.

Another option would be to differentiate whether the feature has been implemented for this specific function at another place, where I have access to the **TableFunction**. The problem here is that the **NotImplementedException**s get thrown at this specific place, so I would exit the parser and throw not implemented within the binder. 

Another option would be to export this logic to a static function like _ordinality_allowed(FunctionExpression fun)_, and then maybe even read the allowed function names from e.g. a ._json_.

What do you think is the way to proceed here?

Reply:
Yea it's not possible to do it here, I think you want to do it in the binder, when we're binding the TableFunction

Reply:
Thanks for the changes! Looks good 👍